### ISSUE TITLE
Possible Load Delay Needed | reveal.js, plot.ly, and Jupyter

### ISSUE THREAD

**Issue by thhancock** [astefanutti/decktape#82]
I have a very graphic intensive Jupyter Notebook that I am exporting to slides via nbconvert.  The slides use  plot.ly charts all throughout.  I have no issues exporting to HTML, converting to slides, opening in a browser to view (as long as there is a ploy.js and reveal.js library in the directory I open the slides in), and / or viewing the slideshow in RISE, but...

When I run decktape on the slides created by nbconvert, the resulting PDF fails to show the plot.ly charts on the slides (all else is fine - i.e. markdown, etc.).  The slides take some time to load when I open them in the browser, so I am surmising that I just need some kind of load delay to occur before the PDF build kicks off.

Is there somewhere in the  code I can add this delay?

**Comment by astefanutti** [comment]
Have you tried the ` --load-pause` option that adds a delay after the presentation has loaded and before decktape start exporting the slides. It's been introduced to address situations similar to yours.

Otherwise, you can use the `--pause` option that adds a delay between each slide.

Finally the main loop is here: https://github.com/astefanutti/decktape/blob/1998908da329142f8ac876a232cb3cb8cade3257/decktape.js#L296 so we may add a delay wherever needed.

**Comment by thhancock** [comment]
Wow.  Glad it did not bite me!

I added the pause at the slide and overall level to no avail.

Had anyone had success with plot.ly charts, reveal, and decktape?

**Comment by astefanutti** [comment]
That may be RISE that complicates things. We could check whether plot.ly charts render correctly in plain reveal.js presentations.

Would you be able to produce a minimal presentation that can be use to reproduce the issue?

**Comment by thhancock** [comment]
Hrm...

I actually just got it working with RISE instead of reveal by launching an ipynb and just pointing it at the localhost with the RISE command.  That got me thinking...

I was trying to run decktape on a static HTML file in the decktape directory (sitting next to my reveal.js and plot.ly files).  Since RISE worked with a locally hosted URL, I just ran a jupyter nbconvert on my notebook and served the HTML.  Then I ran decktape on that URL, and viola!

**Comment by astefanutti** [comment]
Ok, glad it works now. Let me know if you face any other issues.

**Comment by matthewsmithgls** [comment]
I am having the same issue as thhancock when i try to convert using nbconvert the graphics from my plot.ly charts do not show up in the PDF.  

Testing different charts and pauses led me to the following.  The pause did not make a difference but the type of chart did.  I get graphics if I try to convert a line chart to PDF but if i run the exact same code only changing the chart to a bar nothing shows up in the PDF.  Also if i had both a line and a bar chart as separate slides depending on which chart is the first slide I either get nothing (bar chart first) or the line chart shows and the bar chart does not (line chart first).  

I have included the PDF, slides and the notebook/code I used. any help would be greatly appreciated!

[decktape data.zip](https://github.com/astefanutti/decktape/files/735777/decktape.data.zip)
